### ISSUE TITLE
get_cell_tile_data not work correctly?

### ISSUE THREAD

**Issue by Torguen** [godotengine/godot#88510]
### Tested versions

v4.3.dev3.official [36e943b6b]

### System information

w10 64

### Issue description

I have a tile at coordinate 0.0 flipped horizontally.
When I get its tiledata it tells me that the tile is not flipped horizontally.
get_cell_tile_data not working correctly?
All in layer 0

Tile flipped horizontally

![Captura5](https://github.com/godotengine/godot/assets/19652075/3b7300f1-0448-480b-9445-f0fef6224045)

The code gets the tiledata object for that cell but this object indicates that the tile is not flipped horizontally

![Captura5](https://github.com/godotengine/godot/assets/19652075/5ec8a3e4-7ad5-44f6-9b78-f30e4f3b4c4f)

![Captura5](https://github.com/godotengine/godot/assets/19652075/d5b5f0ed-7ab3-4efc-933f-b8c4e900b73f)

### Steps to reproduce

Simply put a tile flipped horizontally
get his tiledata object and write is value flip_h in console

### Minimal reproduction project (MRP)

[tets_tilemap.zip](https://github.com/godotengine/godot/files/14323680/tets_tilemap.zip)

**Comment by AThousandShips** [comment]
Does it work if you do `await get_tree().process_frame` before checking?

**Comment by Torguen** [comment]
No, false is writed

func _ready():
	await get_tree().process_frame
	print( get_cell_tile_data( 0, Vector2i( 0, 0 ) ).flip_h )

**Comment by AThousandShips** [comment]
And `print(get_cell_tile_data(0, Vector2i(0, 0) true).flip_h)`

**Comment by Torguen** [comment]
All false

func _ready():
	await get_tree().process_frame
	print( get_cell_tile_data( 0, Vector2i( 0, 0 ) ).flip_h )
	print( get_cell_tile_data(0, Vector2i(0, 0), true).flip_h )

**Comment by Torguen** [comment]
A comment a little far from this.
How do I apply a tiledata object to a cell?
Tilemap does not have a "set_cell_tile_data" function.

**Comment by AThousandShips** [comment]
You can't, you set the tile from the tile set, the tile data is more complex, you use things like [`set_cell`](https://docs.godotengine.org/en/latest/classes/class_tilemap.html#class-tilemap-method-set-cell)

The data is set *per tile* in the `ToleSet`, not per cell

**Comment by kleonc** [comment]
Since #80144 individual cell tile transformations are encoded within cell's alternative tile id, they're a part of the `alternative_tile` parameter of [`TileMap.set_cell`](https://docs.godotengine.org/en/4.2/classes/class_tilemap.html#class-tilemap-method-set-cell) (it's supported only for tiles from TileSetAtlasSource though).

The per-tile transformations for actual TileSet's tiles (including actual alternative tiles) which are stored within the TileData of course still work, but they're a separate thing.

This needs better documentation. Currently it seems to be mentioned only in the description of the [`TileSetAtlasSource.TRANSFORM_*` constants](https://docs.godotengine.org/en/4.2/classes/class_tilesetatlassource.html#constants).

**Comment by wojtossfm** [comment]
I came here looking for an answer and happily spotted it in https://github.com/godotengine/godot/issues/88510#issuecomment-1951419871 but felt it is worth leaving a comment adding a more verbose comment explaining it.

In 4.3 on a TileMapLayer in a script you can use the following snippet to get the flags for a given tilemap cell
```
	for cell_id_t in self.get_used_cells_by_id():
		var cell_id: Vector2i = cell_id_t
		var alternative_cell_id: int = self.get_cell_alternative_tile(cell_id)

		var flipped_h: bool =  alternative_cell_id & TileSetAtlasSource.TRANSFORM_FLIP_H
		var flipped_v: bool = alternative_cell_id & TileSetAtlasSource.TRANSFORM_FLIP_V
		var transposed: bool = alternative_cell_id & TileSetAtlasSource.TRANSFORM_TRANSPOSE
```

I didn't yet delve into how if at all this would need to be combined with the values from TileData but works for my straight-forward use-case.

**Comment by kleonc** [comment]
> In 4.3 on a TileMapLayer in a script you can use the following snippet to get the flags for a given tilemap cell

Note that in 4.4 there are `TileMap.is_cell_{flipped_h|flipped_v|transposed}` helper methods added (in #81973).

> I didn't yet delve into how if at all this would need to be combined with the values from TileData but works for my straight-forward use-case.

For rendering[^1] they're simply XORed with each other, so e.g.:

```gdscript
var tile_flip_h: bool = get_cell_tile_data(cell_id).flip_h
var cell_flip_h: bool = get_cell_alternative_tile(cell_id) & TileSetAtlasSource.TRANSFORM_FLIP_H
#var cell_flip_h: bool = is_cell_flipped_h(cell_id) # 4.4+

var combined_flip_h: bool = tile_flip_h != cell_flip_h
```

https://github.com/godotengine/godot/blob/2582793d408ade0b6ed42f913ae33e7da5fb9184/scene/2d/tile_map_layer.cpp#L2316-L2329

[^1]: For navigation/physics shapes only cell-flags are applied, as in TileSet each alternative tile have its own TileData with its own shapes created by the user (so already should be matching the flipped/transformed visuals presented to the user in the TileSet editor).
